### PR TITLE
Support custom Python version when build "demohub" image for testing

### PR DESCRIPTION
**Pull request by remoteeng00**
- Support custom Python version when build demohub for testing
- Fix some minor bugs on tests

### CODE REVIEW

**Comment by newgene** [comment]
@remoteeng00 maybe use

```
sudo apt-get install software-properties-common
# adding python repository 
sudo add-apt-repository ppa:deadsnakes/ppa
sudo apt update
# install python 3.7
sudo apt install python3.7
```
[ref] https://towardsdatascience.com/installing-multiple-alternative-versions-of-python-on-ubuntu-20-04-237be5177474

to install different python versions, instead of building it from scratch?

**Comment by newgene** [comment]
also maybe we can upgrade to use 22.04 as the base image now (and node:16 or node:18 for build-webapp)?

and ES8 as the default ES, but support ES7 for testing?

**Comment by remoteeng00** [comment]
> sudo apt-get install software-properties-common
>  .#adding python repository 
> sudo add-apt-repository ppa:deadsnakes/ppa
> sudo apt update
> .# install python 3.7
> sudo apt install python3.7

I did see this solution, but I did not find the python3.10 in the supported Python versions https://launchpad.net/~deadsnakes/+archive/ubuntu/ppa
So I though the Python3.10 is not available on Ubuntu 20.04. Then I ended up by using pyenv package: https://github.com/yyuu/pyenv.git.

**Comment by remoteeng00** [comment]
> also maybe we can upgrade to use 22.04 as the base image now (and node:16 or node:18 for build-webapp)?

I think it's good to go, let me try to update Dockerfile with Ubuntu 22.04 `ppa:deadsnakes/ppa` package and see how it works.

> and ES8 as the default ES, but support ES7 for testing?

It's ok, but we still have some failed test cases on ES8, so I need to update fixture data to pass those test cases.

File: Dockerfile [hunk 1]
```diff
@@ -91,6 +95,13 @@ RUN apt-get -qq -y update && \
         tzdata \
         python3 \
         net-tools \
+        build-essential \
+        libssl-dev \
+    	# must use libssl1.1 when install mongodb on Ubuntu 22.04 \
+        libssl1.1 \
+        libffi-dev \
+        libsqlite3-dev \
+        liblzma-dev \
```
Comment (newgene) [inline Dockerfile:104]: I believe some of these packages are needed when we need to build a custom Python version, except for `libssl1.1` for mongodb, so we should move them to `RUN if [ -n "$PYTHON_VERSION" ]; ` block? This helps to reduce the image size when we just use the default Python from ubuntu.

**Pull request merged**